packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1180,6 +1193,21 @@ export function purposeServiceBuilder(
 
       const purposeToClone = await retrievePurpose(purposeId, readModelService);
 
+      const activeConsumerDelegation = purposeToClone.data.delegationId
+        ? await retrieveDelegation(
+            purposeToClone.data.delegationId,
+            readModelService
+          )
+        : undefined;
+
+      if (
+        activeConsumerDelegation &&
+        activeConsumerDelegation.delegateId === organizationId &&
+        purposeToClone.data.consumerId !== organizationId
+      ) {
```

Comment:
This does not seem to implement what the requirement states (or maybe I'm missing something -- but still I find it hard to read): https://pagopa.atlassian.net/wiki/spaces/PDNDI/pages/1236959344/SRS+Incaricato#Clonazione-finalit%C3%A0-in-delega

I would reuse existing validators and do something like:
- If the purpose has the activeConsumerDelegation it means that it was created by the delegate >>> in that case, it can be cloned both by the delegate and by the delegator, so you can try `assertRequesterIsConsumer` and it fails you catch and try `assertRequesterIsConsumerDelegate`
- If the purpose does not have a delegationId then the purpose was created by the delegator so it means that only the delegator can clone it, you can call `assertRequesterIsConsumer`